src/main/java/org/elasql/procedure/tpart/TPartStoredProcedureTask.java

```diff
@@ -108,4 +142,20 @@ public ProcedureType getProcedureType() {
 	public boolean isReadOnly() {
 		return tsp.isReadOnly();
 	}
+
+	public void setStartTime(long txStartTime, long sinkStartTime, long sinkStopTime, long threadInitStartTime) {
+//		this.firstTxStartTime = firstTxStartTime;
+		this.txStartTime = txStartTime;
+		this.sinkStartTime = sinkStartTime;
+		this.sinkStopTime = sinkStopTime;
+		this.threadInitStartTime = threadInitStartTime;
+	}
+	
+	public static void setFirstTxStartTime(long firstTxStartTime) {
```

Comment:
Static methods should be also put before all other methods along with static variables. Please move this method to the lines right after `private static long firstTxStartTime;`. Like:

```java
    VanillaDb.taskMgr().runTask(new WaitingForStartingRecordTask());
}

private static long firstTxStartTime;

public static void setFirstTxStartTime(long firstTxStartTime) {
   // Your code
}

private TPartStoredProcedure<?> tsp;
private int clientId, connectionId, parId;
```